src/components/Search/index.tsx

```diff
@@ -251,7 +206,9 @@ function Search({queryJSON, isCustomQuery}: SearchProps) {
                 return;
             }
 
-            setSelectedTransactions(prepareTransactionsList(item, selectedTransactions));
+            const transactionList = prepareTransactionsList(item, selectedTransactions);
+            setSelectedTransactions(transactionList);
+            setSelectedReports(SearchUtils.getReportsFromSelectedTransactions(data, transactionList));
```

Comment:
since we are always calling `setSelectedReports` together with `setSelectedTransactions` - just create 1 function that is called once, with both arguments